mne/decoding/transformer.py

```diff
@@ -79,10 +83,16 @@ def fit(self, epochs_data, y):
 
         for key, this_pick in picks_list.items():
             if self.with_mean:
-                ch_mean = X[:, this_pick, :].mean(axis=1)[:, None, :]
+                if len(this_pick) == 0:
+                    ch_mean = np.nan * np.ones((X.shape[0], 1, X.shape[2]))
```

Comment:
`self.std_` is a public attribute that always has a value for `mag`, `grad`, and `eeg`. This gives it the same value it was getting before, but does it without taking the mean or std of an empty slice